LeaderboardBackend.Test/Categories.cs

```diff
@@ -288,4 +289,346 @@ public static async Task CreateCategory_BadData(string? name, string? slug, obje
+    {
+        ExceptionAssertions<RequestFailureException> exAssert = await FluentActions.Awaiting(() => _apiClient.Delete(
+            $"/category/-1",
+            new()
+            {
+                Jwt = _jwt,
+            }
+        )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
+
+        ConflictDetails<CategoryViewModel>? problemDetails = await exAssert.Which.Response.Content.ReadFromJsonAsync<ConflictDetails<CategoryViewModel>>(TestInitCommonFields.JsonSerializerOptions);
+        problemDetails!.Title.Should().Be("Not Found");
+    }
+
+    [Test]
+    public static async Task DeleteCategory_NotFound_AlreadyDeleted()
+    {
+        ApplicationContext context = _factory.Services.CreateScope().ServiceProvider.GetRequiredService<ApplicationContext>();
+
+        Category cat = new()
+        {
+            Name = "Deleted",
+            Slug = "deletedcat-already-deleted",
+            LeaderboardId = _createdLeaderboard.Id,
+            SortDirection = SortDirection.Ascending,
+            Type = RunType.Score,
+            DeletedAt = _clock.GetCurrentInstant(),
+        };
+
+        context.Categories.Add(cat);
+        await context.SaveChangesAsync();
+        cat.Id.Should().NotBe(default);
+
+        ExceptionAssertions<RequestFailureException> exAssert = await FluentActions.Awaiting(() => _apiClient.Delete(
+            $"/category/{cat.Id}",
+            new()
+            {
+                Jwt = _jwt,
+            }
+        )).Should().ThrowAsync<RequestFailureException>().Where(e => e.Response.StatusCode == HttpStatusCode.NotFound);
+
+        ConflictDetails<CategoryViewModel>? problemDetails = await exAssert.Which.Response.Content.ReadFromJsonAsync<ConflictDetails<CategoryViewModel>>(TestInitCommonFields.JsonSerializerOptions);
+        problemDetails!.Title.Should().Be("Already Deleted");
+    }
+
+    [Test]
+    public static async Task RestoreCategory_OK()
+    {
+        CategoryViewModel cat = await _apiClient.Post<CategoryViewModel>(
+            $"leaderboard/{_createdLeaderboard.Id}/categories/create",
+            new()
+            {
+                Body = new CreateCategoryRequest()
+                {
+                    Name = "Restore OK",
+                    Slug = "restorecat-ok",
+                    SortDirection = SortDirection.Ascending,
+                    Type = RunType.Score,
+                },
+                Jwt = _jwt,
+            }
+        );
+
+        cat.Id.Should().NotBe(default);
+
+        await _apiClient.Delete(
+            $"category/{cat.Id}",
+            new()
+            {
+                Jwt = _jwt,
+            }
+        );
+
+        CategoryViewModel retrieved = await _apiClient.Get<CategoryViewModel>(
+            $"api/category/{cat.Id}",
+            new() { }
+        );
+
+        retrieved.DeletedAt.Should().Be(_clock.GetCurrentInstant());
```

Comment:
We don't care about this because we're not testing the delete endpoint here.

Reply:
Resolving this as this comment will be irrelevant with my changes to the test. I'm directly adding an already-deleted cat to the DB.